src/app/core/components/Sidenav/Sidenav.tsx

```diff
@@ -30,74 +30,134 @@ interface SidenavProps {
+  const isB2BWorkspace = !!useSelector(workspacesSelectors.getSelectedWorkspace);
   const isLoadingCredentials = useAppSelector((state: RootState) => state.workspaces.isLoadingCredentials);
+  const pendingInvitations = useAppSelector((state: RootState) => state.shared.pendingInvitations);
 
-  const onDownloadAppButtonClicked = (): void => {
-    const getDownloadApp = async () => {
-      const download = await desktopService.getDownloadAppUrl();
-      return download;
-    };
-    getDownloadApp()
-      .then((download) => {
-        window.open(download, '_self');
-      })
-      .catch(() => {
-        notificationsService.show({
-          text: 'Something went wrong while downloading the desktop app',
-          type: ToastType.Error,
-        });
-      });
-  };
+  const itemsNavigation = [
+    {
+      to: '/',
+      isActive: isActiveButton('/') || isActiveButton('/file/:uuid') || isActiveButton('/folder/:uuid'),
+      label: translate('sideNav.drive'),
+      icon: FolderSimple,
+      iconDataCy: 'sideNavDriveIcon',
+      show: true,
+    },
+    {
+      to: '/backups',
+      isActive: isActiveButton('/backups'),
+      label: translate('sideNav.backups'),
+      icon: ClockCounterClockwise,
+      iconDataCy: 'sideNavBackupsIcon',
+      show: !isB2BWorkspace,
+    },
+    {
+      to: '/shared',
+      isActive: isActiveButton('/shared'),
+      label: translate('sideNav.shared'),
+      icon: Users,
+      notifications: pendingInvitations.length,
+      iconDataCy: 'sideNavSharedIcon',
+      show: true,
+    },
+    {
+      to: '/recents',
+      isActive: isActiveButton('/recents'),
+      label: translate('sideNav.recents'),
+      icon: Clock,
+      iconDataCy: 'sideNavRecentsIcon',
+      show: !isB2BWorkspace,
+    },
+    {
+      to: '/trash',
+      isActive: isActiveButton('/trash'),
+      label: translate('sideNav.trash'),
+      icon: Trash,
+      iconDataCy: 'sideNavTrashIcon',
+      show: true,
+    },
+    {
+      label: translate('sideNav.desktop'),
+      icon: Desktop,
+      iconDataCy: 'sideNavDesktopIcon',
+      onclick: handleDownloadApp,
+      show: !isB2BWorkspace,
+    },
+  ];
 
-  const onLogoClicked = (): void => {
+  const onLogoClicked = () => {
     navigationService.push(AppView.Drive);
   };
 
-  const { subscription, planUsage, planLimit, isLoadingPlanLimit, isLoadingPlanUsage } = props;
-
-  const pendingInvitations = useAppSelector((state: RootState) => state.shared.pendingInvitations);
-
   return (
     <div className="flex w-64 flex-col">
-      {isLoadingCredentials && (
-        <div className="absolute z-50 flex h-full w-full flex-col items-center justify-center bg-highlight/40">
-          <Spinner className="h-10 w-10" />
-          <p className="mt-5 text-2xl font-medium text-gray-100">
-            {translate('workspaces.messages.switchingWorkspace')}
-          </p>
-        </div>
-      )}
+      {isLoadingCredentials && <LoadingSpinner translate={translate} />}
```

Comment:
To make the component dumber you can pass directly the text to be displayed, so it doesn't depend on translate